light/client.go

```diff
@@ -659,12 +662,15 @@ func (c *Client) lightBlockFromPrimary(ctx context.Context) (*types.LightBlock,
 // lightBlockFromPrimaryAtHeight retrieves a light block from the primary provider
 func (c *Client) lightBlockFromPrimaryAtHeight(ctx context.Context, height int64) (*types.LightBlock, error) {
 	c.providerMutex.Lock()
-	l, err := c.primary.LightBlock(ctx, height)
+	l, err := c.getLightBlock(ctx, c.primary, height)
 	c.providerMutex.Unlock()
 
 	switch err {
 	case nil:
-		// Everything went smoothly. We reset the lightBlockRequests and return the light block
+
+	// catch canceled contexts or deadlines
+	case context.Canceled, context.DeadlineExceeded:
```

Comment:
Does this reordering  change anything?

Reply:
I did it 'cause upstream has the same order (just sync)